sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor.cs

```diff
@@ -2065,17 +2059,20 @@ private async Task ValidateEventHubsConnectionAsync(CancellationToken cancellati
             await using var connectionAwaiter = connection.ConfigureAwait(false);
 
             var properties = await connection.GetPropertiesAsync(RetryPolicy, cancellationToken).ConfigureAwait(false);
+            var partitionIndex = new Random().Next(0, (properties.PartitionIds.Length - 1));
```

Comment:
The obvious question here is "why not make the random a static member and instantiate it once?"   

I considered that, but there's really no benefit that I could see for keeping a reference.  We expect applications to have a single instance of the processor in any given process and the only time we query a value is during start up (here) - which is done only once normally and very infrequently in the rare cases that an application stops/restarts the processor.